File: 06_Backtracking/TAEHO/10819.py
from sys import stdin

def backtracking(cnt):
    global max_sum
    if cnt == n:
        max_sum = max(max_sum,sum([abs(back_list[i-1]-back_list[i]) for i in range(1,n)]))

    for i in range(n):
        if visit[i] == 0:
            back_list.append(array[i])
            visit[i] = 1
            backtracking(cnt+1) 
            back_list.pop()
            visit[i] = 0

n = int(stdin.readline())
visit = [0] * n
array = list(map(int,stdin.readline().split()))
back_list = []
max_sum = 0
backtracking(0)
print(max_sum)

File: 06_Backtracking/HONG/10819.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int N, x;
    cin >> N;
    vector<int> arr(N);
    for(int i=0;i<N;i++) cin >> arr[i];
    sort(arr.begin(), arr.end());

    int max_sum = 0;
    do {
        int sum = 0;
        for(int i=0;i<N-1;i++) sum += abs(arr[i] - arr[i+1]);
        max_sum = max(sum, max_sum);
    } while(next_permutation(arr.begin(), arr.end()));

    cout << max_sum;
}